apps/prairielearn/src/lib/url.ts

```diff
@@ -14,3 +18,9 @@ export function getUrl(req: Request): URL {
 export function getSearchParams(req: Request): URLSearchParams {
   return new URL(req.originalUrl, getCanonicalHost(req)).searchParams;
 }
+
+export function getCourseInstanceUrl(
+  context: StudentCourseInstanceContext | StaffCourseInstanceContext,
+): string {
+  return `${config.urlPrefix}/course_instance/${context.course_instance.id}`;
+}
```

Comment:
Three notes:


- Is there a reason this is taking a full context and not either `StaffCourseInstance | StudentCourseInstance` or just `{ id: string }`? The latter would make this much more useful in isolation.
- Is this intended to be used on the client? The usage of the contexts would imply to me that it is, but the `config` here will make that impossible.
- Should this be named `getStudentCourseInstanceUrl` to make it clear that it's that and not an instructor course instance URL?

Reply:
Changed to not use config in [83d3694](https://github.com/PrairieLearn/PrairieLearn/pull/12431/commits/83d3694966c71ee8b7dbce260708a05e6a7fece5) and updated the name in [a5e1f7a](https://github.com/PrairieLearn/PrairieLearn/pull/12431/commits/a5e1f7a37175daa347eca68366fb4389b04aa459)